framework/reactioncommerce/utils/normalize.ts

```diff
@@ -136,17 +142,14 @@ export function normalizeProduct(productNode: CatalogItemProduct): Product {
     slug: slug?.replace(/^\/+|\/+$/g, '') ?? '',
     path: slug ?? '',
     sku: sku ?? '',
-    images: media?.length ? normalizeProductImages(media, title ?? '') : [],
+    images: media?.length ? normalizeProductImages(<ImageInfo[]>media, title ?? '') : [],
     vendor: product.vendor,
     price: {
       value: pricing[0]?.minPrice ?? 0,
       currencyCode: pricing[0]?.currency.code,
     },
-    variants: variants?.length ? normalizeProductVariants(variants) : [],
-    options: variants?.length
-      ? groupProductOptionsByAttributeLabel(variants)
-      : [],
-    ...rest,
+    variants: !!variants ? normalizeProductVariants(<CatalogProductVariant[]>variants) : [],
```

Comment:
We should define a common way for resolving the `Maybe` type because it either causes type check errors or requires redundant type castings.

Reply:
Agreed. This can be part of another PR though.